### PR TITLE
Deduplicate indirect deps with yarn-deduplicate.

### PR DESCRIPTION
**Pull request by gnprice**
Also add a linter (in the form of another tools/test "suite")
to keep it that way.
    
Inspired by zulip/zulip@e57ede942 (zulip/zulip#13118). (thanks @andersk !)

### CODE REVIEW

**Comment by gnprice** [comment]
This script in general is not the awesomest code; I think it's quietly outgrown the scale at which its current style (e.g. a bunch of names that are short and kind of cryptic) is good enough. This branch adds a doc-comment on a function that needed it, but doesn't try to comprehensively address that right now.

Just after writing this, I went and drafted up a branch to start using shellcheck -- inspired by this as well as other recent shell-script-touching PRs. That won't directly help with things like too-short names, but it'll be another step toward better code here.

File: tools/test [hunk 1]
```diff
@@ -162,6 +171,22 @@ run_prettier() {
...
 }
 
+run_deps() {
+    files_check package.json yarn.lock \
+        || return 0
+
+    if ! yarn-deduplicate --fail --list; then
+        cat >&2 <<EOF
+
+Duplicates could be shared in yarn.lock.
```
Comment (ray-kraesig) [inline tools/test]: This error message is more than a little cryptic; I'd suggest something more explicit, along the lines of "`Mergeable duplicate dependencies found in yarn.lock`".

Comment (gnprice) [reply tools/test]: Good idea. I went for even more explicit than that:
> Found duplicate dependencies in yarn.lock which could be dedup'd.

**Comment by gnprice** [comment]
Thanks @ray-kraesig for the review! Merged, with the edit mentioned above.

**Pull request merged**